### PR TITLE
Added ndcg metric

### PR DESCRIPTION
**Pull request by norm4nn**
Added ndcg metric and doctests for it

### CODE REVIEW

File: lib/scholar/metrics/ranking.ex [hunk 1]
```diff
@@ -94,4 +94,42 @@ defmodule Scholar.Metrics.Ranking do
...
+      iex> Scholar.Metrics.Ranking.ndcg_n(true_relevance, true_relevance, [{:k, 4}])
+      #Nx.Tensor<
+        f32
+        1.0
+      >
+  """
+  defn ndcg_n(y_true, y_score, opts) do
+    dcg_n(y_true, y_score, opts) / dcg_n(y_true, y_true, opts)
+  end
+
```
Comment (josevalim) [inline lib/scholar/metrics/ranking.ex:134]:
```suggestion
      iex> Scholar.Metrics.Ranking.ndcg_n(true_relevance, scores, k: 4)
      #Nx.Tensor<
        f32
        0.352024108171463
      >
      iex> Scholar.Metrics.Ranking.ndcg_n(true_relevance, true_relevance, k: 4)
      #Nx.Tensor<
        f32
        1.0
      >
  """
  defn ndcg_n(y_true, y_score, opts) do
    dcg_n(y_true, y_score, opts) / dcg_n(y_true, y_true, opts)
  end
```

File: lib/scholar/metrics/ranking.ex [hunk 2]
```diff
@@ -94,4 +94,42 @@ defmodule Scholar.Metrics.Ranking do
...
+      >
+      iex> Scholar.Metrics.Ranking.ndcg_n(true_relevance, true_relevance, k: 4)
+      #Nx.Tensor<
+        f32
+        1.0
+      >
+  """
+  defn ndcg_n(y_true, y_score, opts) do
+    dcg_n(y_true, y_score, opts) / dcg_n(y_true, y_true, opts)
+  end
```
Comment (msluszniak) [inline lib/scholar/metrics/ranking.ex:134]: I think setting `[]` as default `opts` is more convenient.
```suggestion
      iex> Scholar.Metrics.Ranking.ndcg_n(true_relevance, scores)
      #Nx.Tensor<
        f32
        0.6956940293312073
      >
      iex> scores = Nx.tensor([0.05, 1.1, 1.0, 0.5, 0.0])
      iex> Scholar.Metrics.Ranking.ndcg_n(true_relevance, scores)
      #Nx.Tensor<
        f32
        0.4936802089214325
      >
      iex> scores = Nx.tensor([0.05, 1.1, 1.0, 0.5, 0.0])
      iex> Scholar.Metrics.Ranking.ndcg_n(true_relevance, scores, k: 4)
      #Nx.Tensor<
        f32
        0.352024108171463
      >
      iex> Scholar.Metrics.Ranking.ndcg_n(true_relevance, true_relevance, k: 4)
      #Nx.Tensor<
        f32
        1.0
      >
  """
  defn ndcg_n(y_true, y_score, opts \\ []) do
    dcg_n(y_true, y_score, opts) / dcg_n(y_true, y_true, opts)
  end
```

File: lib/scholar/metrics/ranking.ex [hunk 3]
```diff
@@ -94,4 +94,42 @@ defmodule Scholar.Metrics.Ranking do
         end
     end
   end
+
+  @doc """
+  Computes the NDCG based on true relevance scores (`y_true`) and their respective predicted scores (`y_score`).
```
Comment (krstopro) [inline lib/scholar/metrics/ranking.ex:99]:
```suggestion
Computes the normalized discounted cumulative gain (NDCG) based on true relevance scores `y_true` and their respective predicted scores `y_score`.
```

Let's include the full metric name as well - you never know how people might search for it on hexdocs.

**Review by krstopro — COMMENTED** [review]
Small suggestion regarding docstring.

File: lib/scholar/metrics/ranking.ex [hunk 4]
```diff
@@ -94,4 +94,42 @@ defmodule Scholar.Metrics.Ranking do
...
+
+  @doc """
+  Computes the normalized discounted cumulative gain (NDCG) based on true relevance scores `y_true` and their respective predicted scores `y_score`.
+
+  ## Options
+
+  #{NimbleOptions.docs(@dcg_opts_schema)}
+  
+  ## Examples
+
```
Comment (josevalim) [inline lib/scholar/metrics/ranking.ex:106]:
```suggestion

  ## Options

  #{NimbleOptions.docs(@dcg_opts_schema)}

  ## Examples

```

**Review by msluszniak — APPROVED** [review]

**Pull request merged**